discojs/src/models/onnx.ts

```diff
@@ -0,0 +1,107 @@
+import { AutoModelForCausalLM, PreTrainedModel } from '@xenova/transformers';
+import { Model } from './index.js';
+import type {
+  WeightsContainer
+} from '../index.js';
+import { List } from 'immutable';
+import { Tensor } from '@xenova/transformers';
+
+
+export class ONNXModel extends Model<'text'> {
+  private model: PreTrainedModel;
+
+  private constructor(model: PreTrainedModel) {
+    super();
+    this.model = model;
+  }
+
+  static async init_pretrained(modelName = 'Xenova/gpt2'): Promise<ONNXModel> {
+    const model = await AutoModelForCausalLM.from_pretrained(modelName);
+    return new ONNXModel(model);
+  }
+
+  getConfig(): any {
+    return this.model.config;
+  } 
+
+  async predict(batch: List<List<number>>): Promise<List<number>> {

```

Comment:
When is the `predict` method used? I only see `getLogits` in `computeONNXLogLikelihood`

I think this method should make the assumption that the tokens are already padded/truncated and in the right format, i.e. that `batch` is a list of constant-length lists (as does the GPT class in models/gpt/index.ts). 
The preprocessing should be handled beforehand by the caller of the ONNXModel class. For example for GPT-tfjs it's done in processing/index.ts L.62:
```js
return d.map(text => processing.tokenize(tokenizer, text))
        .flatten()
        .batch(contextLength + 1, 1)
        .map((tokens) => [tokens.pop(), tokens.last()]) as
          Dataset<DataFormat.ModelEncoded[D]>;
```